Source/Fuse.Reactive.JavaScript/ModuleInstance.Reactive.uno

```diff
@@ -0,0 +1,278 @@
+
+			void JSThreadPerform(object dc, int pos)
+			{
+				if (pos == Arguments.Length - SpecialArgCount)
+				{
+					JSThreadPerform(dc);
+					return;
+				}
+
+				var obj = dc as Scripting.Object;
+				if (obj != null)
+				{
+					var key = Arguments[pos].ToString();
+					JSThreadPerform(obj[key], pos+1);
+					return;
+				}
+
+				var arr = dc as Scripting.Array;
+				if (arr != null)
+				{
+					var index = Marshal.ToInt(Arguments[pos]);
+					JSThreadPerform(arr[index], pos+1);
+					return;
+				}
+
+				throw new Error("Unable to update data context. Path doesn't match exports");
+			}
+
+			void UIThreadPerform(object dc, int pos)
+			{
+				if (pos == Arguments.Length - SpecialArgCount)
+				{
+					UIThreadPerform(dc);
+					return;
+				}
+
+				var obj = dc as ObjectMirror;
+				if (obj != null)
+				{
+					var key = Arguments[pos].ToString();
+					UIThreadPerform(obj[key], pos+1);
+					return;
+				}
+
+				var arr = dc as ArrayMirror;
+				if (arr != null)
+				{
+					var index = Marshal.ToInt(Arguments[pos]);
+					UIThreadPerform(arr[index], pos+1);
+					return;
+				}
+
+				throw new Error("Unable to update data context. Path doesn't match exports");
+			}
+		}
+
+		class SetOperation: Operation
+		{
+			public SetOperation(ModuleInstance inst, object[] args): base(inst, args) {}
+
+			object _wrappedValue;
+
+			protected override int SpecialArgCount { get { return 2; } }
+			protected override void JSThreadPerform(object dc)
+			{
+				var key = Arguments[Arguments.Length-2];
+				var value = Arguments[Arguments.Length-1];
+				_wrappedValue = ModuleInstance._worker.Reflect(value);
+
+				var obj = dc as Scripting.Object;
+				if (obj != null) 
+				{
+					obj[key.ToString()] = value;
+					return;
+				}
+
+				var arr = dc as Scripting.Array;
+				if (arr != null) 
+				{
+					arr[Marshal.ToInt(key)] = value;
+					return;
+				}
+
+				throw new Error("module.set(): Path mistmatch");
+			}
+
+			protected override void UIThreadPerform(object dc)
+			{
+				var key = Arguments[Arguments.Length-2];
+
+				var obj = dc as ObjectMirror;
+				if (obj != null) obj.Set(key.ToString(), _wrappedValue);
+
+				var arr = dc as ArrayMirror;
+				if (arr != null) arr.Set(Marshal.ToInt(key), _wrappedValue);
+			}
+		}
+
+		class AddOperation: Operation
+		{
+			public AddOperation(ModuleInstance inst, object[] args): base(inst, args) {}
+
+			object _wrappedValue;
+
+			protected override int SpecialArgCount { get { return 1; } }
+			protected override void JSThreadPerform(object dc)
+			{
+				var value = Arguments[Arguments.Length-1];
+				_wrappedValue = ModuleInstance._worker.Reflect(value);
+
+				var arr = dc as Scripting.Array;
+				if (arr != null)
+				{
+					ModuleInstance._worker.Push(arr, value);
+					return;
+				}
+
+				throw new Error("Path mistmatch");
+			}
+
+			protected override void UIThreadPerform(object dc)
+			{
+				var arr = dc as ArrayMirror;
+				if (arr != null) arr.Add(_wrappedValue);
+			}
```

Comment:
It can be safely assumed to be an ArrayMirror because we are traversing a data structure identical to the one on the JS thread.